### PR TITLE
check and externalize explicit sparql queries #1284

### PR DESCRIPTION
**Pull request by curibe**
related  #1283, #1282, #785 

## Externalize explicit sparql queries   

Converting the explicit queries (tables) first to externalized query format. For example, files who have the following queries:

```
somePanelDescriptionSparql = `
SELECT ...
...
`
``` 
This PR solves partially the issue #1284.

The following files still have explicit queries:

- [x] 404_chemical.html
- [x] author.html
- [x] author-index-curation.html
- [x] authors.html
- [x] award_curation.html
- [x] chemical-index-curation.html
- [x] chemical-index.html
- [x] lexeme_empty.html
- [x] pathway_empty.html
- [x] pathway.html
- [x] property.html
- [x] software_empty.html
- [x] topic_curation.html
- [x] use_empty.html
- [x] use.html
- [x] venue_curation.html
- [x] venues.html
- [x] work_cito.html
- [x] work_empty.html
- [x] works.html

### Some annotations
 - Files with empty will use the aspect as `index` but the name of html file will be the same for the moment. For example,
`use_empty.html` have an aspect named `use-index`. The name of these html will be change in other PR
- Externalization for `author.html`, `pathway.html` and `404_chemical.html` required the creation of  new macros and custom JS functions for extracting queries to external .sparql files

### CODE REVIEW

File: scholia/app/templates/author-index-curation.html [hunk 1]
```diff
@@ -165,32 +33,32 @@ <h2>Author name strings to be resolved</h2>
...
 Wikidata items.
 Follow the link to use the Author disambiguator tool to try to resolve
 the authors.
 
-<table class="table table-hover" id="missing-citing-author-items"></table>
+<table class="table table-hover" id="missing-citing-author-items-table"></table>
 
 
-<h2>Authored works with missing topics</h2>
+<h2 d="missing-topic-header">Authored works with missing topics</h2>
```
Comment (DanielFLopez) [inline scholia/app/templates/author-index-curation.html:59]: maybe it's id?

Comment (fnielsen) [reply scholia/app/templates/author-index-curation.html:59]: yes, that should be id

Comment (curibe) [reply scholia/app/templates/author-index-curation.html:59]: Yes, It's id

**Comment by fnielsen** [comment]
Sorry, about the name pattern of the files not being clear. It should be aspect_panel-description, e.g., `{% set aspect = "chemical-index" %}`. This is confusingly used. So the file is `chemical-index_statistics.sparql`

**Comment by fnielsen** [comment]
`venues` are named fine!

**Comment by fnielsen** [comment]
Files called "empty" should be "index" instead, e.g., "work-index" and "pathway-index". Sorry again

**Comment by fnielsen** [comment]
`award_curation` -> `award-curation`

**Comment by curibe** [comment]
> `award_curation` -> `award-curation`

@fnielsen In this case the html is called "award_curation.html". Do I still assign it in aspect as "award-curation"? no matter if the html file uses "_" or "-"?

**Comment by curibe** [comment]
> Files called "empty" should be "index" instead, e.g., "work-index" and "pathway-index". Sorry again

@fnielsen Do you want to change the name only in the `{%set aspect=....%}` or also the html name?

**Comment by fnielsen** [comment]
> > `award_curation` -> `award-curation`
> 
> @fnielsen In this case the html is called "award_curation.html". Do I still assign it in aspect as "award-curation"? no matter if the html file uses "_" or "-"?

All HTML and sparql files should be called with a dash instead of an underscore

**Comment by curibe** [comment]
@fnielsen please, can you check this commit  11c714837b3934b10f2e884b9082503c843c9bae  where a new macro was implemented togetther with JS functions in `scholia.js`

**Comment by curibe** [comment]
@fnielsen do you also require this caption here?
image

**Comment by fnielsen** [comment]
> @fnielsen do you also require this caption here?

I am not sure what you mean?

**Comment by curibe** [comment]
> > @fnielsen do you also require this caption here?
> 
> I am not sure what you mean?

I refer to the caption shown in the red box. The function that shows the diagram  does not a code implemented  to display the
link, but another caption (below-left in the image :  _This diagram is  showing..._ ):
https://scholia.toolforge.org/pathway/Q28031254
image

I put it to keep the format as the other externalizations. You can see the changes in this commit: a37c6004609e30597fab99868a2858dc726bcad1
The caption was added here:
image
And It looks like this:
image

**Comment by fnielsen** [comment]
No a caption there is not necessary, I think. The rendering does not come from Wikimedia services, but from a thirdparty service, available at https://pathway-viewer.toolforge.org/?id=WP111 . I do not think they construct the visualization with SPARQL.

**Comment by fnielsen** [comment]
* cito_empty should be cito-index
* is venues iframe panels missing? http://127.0.0.1:8100/venues/Q1981225,Q5936947

**Pull request merged**